consonance/ml_logic/omr.py

```diff
@@ -4,35 +4,69 @@
 
 import numpy as np
 import matplotlib.pyplot as plt
+import pandas as pd
 
 from pathlib import Path
 
 import tensorflow as tf
 import keras
 from keras import ops
 from keras import layers
-
+from keras.preprocessing.sequence import pad_sequences
 
 
 # Path to the data directory
```

Comment:
It's best if executable code is wrapped in a function in a python code. In the notebook you can run all cells from top to bottom, but it works a bit differently in a web app.

Constants or environment variables can be declared outside of a function scope and can be imported/accessed globally/within a file scope, but executable code which relies on computed variables can break an app build if the values needed are not available in the current state. Make sure you have access to these variables and their values are not empty at all times the code is accessed by the app.

Ex. max_length variable broke the build because labels were not available.
In cases like this it's safer if this code is included inside of a function, and checks are in place to see if our data has the desired type and value.